### PR TITLE
support dot in env variable names [e.g. POSTGRESQL.BASE.USER]

### PR DESCRIPTION
**Pull request by flamewow**
Supporting dots in environment variable names, like 

```
POSTGRESQL.BASE.USER=postgres
POSTGRESQL.MAIN.USER=${POSTGRESQL.BASE.USER}
```

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -24,12 +24,12 @@ function _interpolate (envValue, environment, config) {
    * (
    *   (?!(?<=\\))\$        // only match dollar signs that are not escaped
    *   {?                   // optional opening curly brace
-   *     ([\w]+)            // match the variable name
+   *     ([\w.]+)            // match the variable name
```
Comment (FezVrasta) [inline lib/main.js:27]:
```suggestion
   *     ([\w.]+)           // match the variable name
```

**Review by FezVrasta — APPROVED** [review]

File: lib/main.js (hunk 1 above)
Comment (FezVrasta) [inline lib/main.js:27]:
```suggestion
   *     ([\w.]+)           // match the variable name
```

**Comment by motdotla** [comment]
for sanity we are not going to support this. the [`dotenv`](https://github.com/motdotla/dotenv) lib does not nor do other popular dotenv implementations like python's.

why not just replace the `.` with a `_`. do you not have control over the variable name? you certainly do inside your `.env` file. do you not when going to production?

**Comment by flamewow** [comment]
@motdotla 
In a big project, a change like that (going from "." to "_") is going to require a cascade of updates.
And it a pain in the ass to get those name changed.

**Comment by motdotla** [comment]
> btw, dotenv lib loads such env variables no propblem

you're right. what was i thinking. it can be accessed via `process.env['POSTGRESQL.BASE.USER']`

let me dig deeper into what other implementations are supporting.

also, can someone give some more color on why you are naming your variables this way.

**Comment by motdotla** [comment]
i think i'm wrong here and @flamewow and others are correct. following the principle of least surprise, this should just work. merging.

**Comment by flamewow** [comment]
> > btw, dotenv lib loads such env variables no propblem
> 
> you're right. what was i thinking. it can be accessed via `process.env['POSTGRESQL.BASE.USER']`
> 
> let me dig deeper into what other implementations are supporting.
> 
> also, can someone give some more color on why you are naming your variables this way.

Legacy code, otherwise I'd changed them to a format that works.

**Pull request merged**